web_ui/src/shared/components/header/jobs-management/jobs-dialog.component.tsx

```diff
@@ -83,18 +113,26 @@ export const JobsDialog = ({ isFullScreen, onClose, setIsFullScreen }: JobsDialo
         }
     );
 
-    const isInitialRange = isEqual(range, INITIAL_DATES);
+    const isInitialRange = isEmpty(range);
 
     const areFiltersChanged = !isInitialRange || !!filters.projectId || !!filters.userId || !!filters.jobTypes.length;
 
     const allJobs = getAllJobs(data);
+
+    jobsCountRef.current = handleDataCountWhenScheduledShouldNotBeFiltered(
+        jobsCountRef.current,
+        data?.pages?.at(0)?.jobsCount,
+        !isEmpty(range),
+        selectedJobState !== JobState.SCHEDULED
+    );
+
     const {
         numberOfRunningJobs,
         numberOfFinishedJobs,
         numberOfScheduledJobs,
         numberOfCancelledJobs,
         numberOfFailedJobs,
-    } = data?.pages?.at(0)?.jobsCount ?? DEFAULT_JOBS_COUNT;
+    } = jobsCountRef.current;
```

Comment:
One part that worries me slightly is that the react docs [recommend not to read/write from refs during renders](https://react.dev/learn/referencing-values-with-refs#best-practices-for-refs).
In this case this probably is fine since we are setting the ref right above at line 121. But let's keep an eye out in case we make any more changes to this component.